projects/igniteui-angular-elements/src/analyzer/printer.ts

```diff
@@ -0,0 +1,132 @@
+import * as ts from 'typescript';
+import * as path from 'node:path';
+import * as fs from 'node:fs';
+import { format } from 'prettier';
+import type { ComponentMetadata, ContentQuery } from './types';
+
+
+export class AnalyzerPrinter {
+
+    #program: ts.Program;
+    #checker: ts.TypeChecker;
+    #printer: ts.Printer;
+    #configPath: string;
+    #configFile: ts.SourceFile;
+
+    // TODO: Change to the default output path
+    public out = 'elements.config.ts';
+
+    constructor(public config = 'projects/igniteui-angular-elements/src/analyzer/config.template.ts') {
+        this.#configPath = path.join(process.cwd(), config);
+        this.#program = ts.createProgram([this.#configPath], {});
```

Comment:
Unless TypeScript optimizes this behind the scenes (almost certain it doesn't cuz it crating this takes a bit and afterwards again), this could also be a point of optimization - I think it should be possible to use the config file + main entry for the lib to create a single program that has the it and can resolve the entire lib source.